tests/support/util.tcl

```diff
@@ -599,6 +599,29 @@ proc stop_bg_complex_data {handle} {
     catch {exec /bin/kill -9 $handle}
 }
 
+# Execute a background process doing xadd for the specified number.
+proc start_bg_xadd {host port db ops} {
+    set tclsh [info nameofexecutable]
+    exec $tclsh tests/helpers/bg_xadd.tcl $host $port $db $ops $::tls &
+}
+
+# Stop a process executed with start_bg_xadd.
+proc stop_bg_xadd {handle} {
+    catch {exec /bin/kill -9 $handle}
+}
+
+# Execute a background process doing xreadgroup block for the specified number.
+proc start_bg_block_xreadgroup {host port db ops} {
+    set tclsh [info nameofexecutable]
+    exec $tclsh tests/helpers/bg_block_xreadgroup.tcl $host $port $db $ops $::tls &
+}
+
+# Stop a process executed with start_bg_block_xreadgroup.
+proc stop_bg_block_xreadgroup {handle} {
+    catch {exec /bin/kill -9 $handle}
+}
```

Comment:
not a fun of adding all these "generic" helpers for one little test (unlikely that they'll be re-usable).
instead, i hope that we can trigger the problem with a very specific test that targets it.
e.g., pause the server (SIGSTOP) if needed, write some pipeline of commands in a specific order, and hope they'll get processed in just the right way to trigger that infinite loop.

maybe it's even simpler. correct me if i'm wrong, all we need is 3 consumers (3 calls to XREADGROUP, without any retry loop), and one XADD to wake one of them and cause the other two to go into an endless loop.
am i wrong?

Reply:
I overcomplicated the problem... originally, I wanted to get something like bg_block_op to test the stream block
you are right, i updated the code. one word awakens the dreamer. thanks for the review.